### PR TITLE
Updating your SDK via git 

### PR DESCRIPTION
**Pull request by shaunsingh**
Fixes https://github.com/gamemanual0/gm0/issues/113, 

Todo: fix linting, fix images

### CODE REVIEW

File: source/docs/software/using-android-studio.rst [hunk 1]
```diff
@@ -331,6 +331,50 @@ Installing SDK Tools
 
 To install any SDK tools such as platform tools or build tools, open the SDK manager and go to SDK Tools. Select the tools you want to install and install them the same way you would for the SDK platforms.
 
+
+Updating the SDK using Git
+--------------------------
+
+The SDK can be updated through the use of git and version control. In order to update the SDK, we will first need to set the source of the updates through git upstream, and then merge the updates into our current project.
```
Comment (dansman805) [inline source/docs/software/using-android-studio.rst:338]:
```suggestion
The SDK can be updated through the using the version control software Git. In order to update the SDK, we will first need to set the source of the updates using Git remotes, and then merge the updates into our current project.
```

File: source/docs/software/using-android-studio.rst [hunk 2]
```diff
@@ -331,6 +331,50 @@ Installing SDK Tools
...
+
+.. note:: Your FTC project needs to be set up as a Github Repository prior to following this guide.
```
Comment (dansman805) [inline source/docs/software/using-android-studio.rst:340]:
```suggestion
.. note:: Your FTC project needs to be set up as a Git repository prior to following this guide.
```

File: source/docs/software/using-android-studio.rst [hunk 3]
```diff
@@ -331,6 +331,50 @@ Installing SDK Tools
...
+
+Setting git upstream
+^^^^^^^^^^^^^^^^^^^^
```
Comment (dansman805) [inline source/docs/software/using-android-studio.rst:343]:
```suggestion
Setting Git Remote
^^^^^^^^^^^^^^^^^^
```

File: source/docs/software/using-android-studio.rst [hunk 4]
```diff
@@ -331,6 +331,50 @@ Installing SDK Tools
...
+
+#. Open up your project in android studio
```
Comment (dansman805) [inline source/docs/software/using-android-studio.rst:345]:
```suggestion
#. Open up your project in Android Studio
```

File: source/docs/software/using-android-studio.rst [hunk 5]
```diff
@@ -331,6 +331,50 @@ Installing SDK Tools
...
+
+Updating the SDK
+^^^^^^^^^^^^^^^^
+
+In order to update the SDK, run ``git pull upstream master``
+
+Merge Conflicts
+^^^^^^^^^^^^^^^
+
+Merge conflicts occur when competing changes are made to the same line of a file. If you modified gradle files within your project, it is very likely you will run into a merge conflict when updating the sdk.
```
Comment (dansman805) [inline source/docs/software/using-android-studio.rst:359]:
```suggestion
Merge conflicts occur when competing changes are made to the same line of a file. If you modified Gradle files within your project, it is very likely you will run into a merge conflict when updating the SDK.
```

**Review by dansman805 — CHANGES_REQUESTED** [review]

**Review by dansman805 — APPROVED** [review]

**Comment by NoahAndrews** [comment]
The other thing that this should probably explain is how to start the git project from the official repository. If you just start a brand new history, these instructions won't work as expected.

**Comment by shaunsingh** [comment]
The recommended way would be to fork the project and set the remote to ,correct

File: source/docs/software/using-android-studio.rst [hunk 6]
```diff
@@ -331,6 +331,50 @@ Installing SDK Tools
...
+.. note:: Your FTC project needs to be set up as a Git repository prior to following this guide.
+
+Setting Git Remote
+^^^^^^^^^^^^^^^^^^
+
+#. Open up your project in Android Studio
+#. Open up the "terminal" tab
+#. Paste in ``git remote add upstream https://github.com/FIRST-Tech-Challenge/FtcRobotController/`` and hit enter
+
+.. note:: The command above assumes you're using the FTC SDK. If you are using a modified version of the SDK as your base (e.g. Roadrunner Quickstart), replace the git link with one from the modified SDK repo
```
Comment (dansman805) [inline source/docs/software/using-android-studio.rst:349]:
```suggestion
.. note:: The command above assumes you're using the FTC SDK. If you are using a modified version of the SDK as your base (e.g. Road Runner Quickstart), replace the git link with one from the modified SDK repo
```

**Comment by dansman805** [comment]
> The recommended way would be to fork the project and set the remote to ,correct

The recommended way is to clone the project, and then if using GitHub, making a new repository there and pushing there. GitHub forks are meant for submitting pull requests.

**Comment by shaunsingh** [comment]
If possible, could someone kindly write how to upload to GitHub using the Cli? I don't have any experience myself.

**Comment by shaunsingh** [comment]
fixes https://github.com/gamemanual0/gm0/issues/111

**Comment by Eeshwar-Krishnan** [comment]
I think all that needs to be done is merge this with the current gm0 to clear merge conflicts?

**Comment by Eeshwar-Krishnan** [comment]
closing as out of scope after some discussion, teaching the basics of how to use git and deal with merge conflicts quickly becomes way out of scope for basic use

**Comment by abidingabi** [comment]
For slightly more context, if you want to explain updating the SDK, you have to cover:
- cloning
- staging
- commiting
- remotes
- pulling
- merge conflicts
- probably also at this point pushing in order to collaborate/backup your code

Yes gm0 *could* cover this but if it'll end up being a decently fully-featured git tutorial plenty of those already exist

**Pull request closed without merging**